src/main/java/com/esri/samples/featurelayers/statistical_query_group_and_sort/StatisticalQueryGroupAndSortController.java

```diff
@@ -0,0 +1,274 @@
+
+  public ComboBox<String> fieldNameComboBox;
+  public ComboBox<String> statisticTypeComboBox;
+  public TableView<StatisticDefinition> statisticDefinitionsTableView;
+  public TreeView<String> statisticRecordTreeView;
+  public ListView<GroupField> groupFieldsListView;
+  public TableColumn<StatisticDefinition, String> statisticDefinitionFieldNameTableColumn;
+  public TableColumn<StatisticDefinition, StatisticType> statisticDefinitionStatisticTypeTableColumn;
+  public Button removeStatisticDefinitionButton;
+  public TableView<OrderByField> orderByTableView;
+  public TableColumn<OrderByField, String> orderByFieldNameTableColumn;
+  public TableColumn<OrderByField, QueryParameters.SortOrder> orderBySortOrderTableColumn;
+  public Button addOrderByFieldButton;
+  public Button removeOrderByFieldButton;
+
+  private ServiceFeatureTable featureTable;
+
+  public void initialize() {
+
+    setUpButtonBindings();
+    initializeStatisticDefinitionsTableView();
+    initializeGroupByListView();
+    initializeOrderByTableView();
+
+    // load the service feature table
+    String USStatesServiceUri = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Census/MapServer/3";
+    featureTable = new ServiceFeatureTable(USStatesServiceUri);
+    featureTable.loadAsync();
+    featureTable.addDoneLoadingListener(() -> {
+      if (featureTable.getLoadStatus() == LoadStatus.LOADED) {
+        // populate the field name combo box with the feature table's field names
+        List<String> fieldNames = featureTable.getFields().stream().map(Field::getName).collect(Collectors.toList());
+        fieldNameComboBox.getItems().addAll(fieldNames);
+
+        // populate the statistic type combo box with the different StatisticType values
+        statisticTypeComboBox.getItems().addAll(Arrays.stream(StatisticType.values()).map(Enum::toString).collect(Collectors.toList()));
+
+        // populate the Group By list with all of the field names with none of the check boxes selected
+        groupFieldsListView.getItems().addAll(fieldNames.stream().map(f -> new GroupField(f, false)).collect
+            (Collectors.toList()));
+
+        // set some default selections
+        fieldNameComboBox.getSelectionModel().select("AGE_5_17");
+        statisticTypeComboBox.getSelectionModel().select("MINIMUM");
+        statisticDefinitionsTableView.getItems().addAll(
+            new StatisticDefinition("POP2007", StatisticType.SUM),
+            new StatisticDefinition("POP2007", StatisticType.AVERAGE),
+            new StatisticDefinition("AGE_5_17", StatisticType.MINIMUM)
+        );
+        groupFieldsListView.getItems().stream().filter(f -> f.getFieldName().equals("SUB_REGION")).collect(Collectors
+            .toList()).get(0).setGrouping(true);
+        orderByTableView.getItems().add(new OrderByField(new QueryParameters.OrderBy("SUB_REGION", QueryParameters.SortOrder
+            .ASCENDING)));
+      }
+    });
```

Comment:
Would place an alert here if the feature table doesn't load.
new Alert(Alert.AlertType.ERROR, "").show();